core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/salt/update/SaltUpdateActions.java

```diff
@@ -116,16 +135,48 @@ protected Selectable createRequest(ClusterViewContext context) {
     @Bean(name = "SALT_UPDATE_FAILED_STATE")
     public Action<?, ?> saltUpdateFailedAction() {
         return new AbstractStackFailureAction<ClusterCreationState, ClusterCreationEvent>() {
+
+            @Inject
+            private StackDtoService stackDtoService;
+
             @Override
             protected void doExecute(StackFailureContext context, StackFailureEvent payload, Map<Object, Object> variables) {
-                saltUpdateService.handleClusterCreationFailure(context.getStack(), payload.getException());
+                Boolean checkIfBackupDatalakeDatabaseFlowChain = isBackupDatalakeDatabaseFlowChain(payload.getResourceId());
+                saltUpdateService.handleClusterCreationFailure(context.getStack(), payload.getException(), context.getStackId(),
+                        checkIfBackupDatalakeDatabaseFlowChain);
                 sendEvent(context);
             }
 
+            @Override
+            protected StackFailureContext createFlowContext(FlowParameters flowParameters,
+                StateContext<ClusterCreationState, ClusterCreationEvent> stateContext, StackFailureEvent payload) {
+                if (isBackupDatalakeDatabaseFlowChain(payload.getResourceId())) {
+                    StackView stack = stackDtoService.getStackViewById(payload.getResourceId());
+                    MDCBuilder.buildMdcContext(stack);
+                    Map<Object, Object> variables = stateContext.getExtendedState().getVariables();
+                    ProvisionType provisionType = (ProvisionType) variables.getOrDefault(PROVISION_TYPE, ProvisionType.REGULAR);
+                    return new StackFailureContext(flowParameters, stack, stack.getId(), provisionType);
+                } else {
+                    return (super.createFlowContext(flowParameters, stateContext, payload));
+                }
+            }
```

Comment:
in general i do not think it is a good approach to add logic to SaltUpdateActions which belongs to a completely different feature

Reply:
Valid point, but we do not see any other way to do this if we are to proceed with this. We will, of course, remove this as soon as we can find a way to avoid the Salt update process altogether.